src/term/mod.rs

```diff
@@ -65,14 +71,23 @@ impl<'a> RenderableCellsIter<'a> {
             line: Line(0),
             column: Column(0),
             selection: selection,
+            custom_cursor_colors: custom_cursor_colors,
+            undercursorfg: Color::Spec(Rgb { r: 0, g: 0, b: 0 }),
+            undercursorbg: Color::Spec(Rgb { r: 0, g: 0, b: 0 }),
         }.initialize()
     }
 
-    fn initialize(self) -> Self {
+    fn initialize(mut self) -> Self {
         if self.cursor_is_visible() {
-            self.grid[self.cursor].swap_fg_and_bg();
-        }
+            if self.custom_cursor_colors {
+                let (fg, bg) = self.grid[self.cursor].set_cursor();
```

Comment:
done. this is super clean, and i was able to kill the swap in the Cell too, and move it here